src/components/playback/StreamPlayer.js

```diff
@@ -81,20 +82,55 @@ export default class StreamPlayer extends React.Component {
     this.audio.pause()
   }
 
+  toggleFullscreen () {
+    if (this.state.fullscreen) {
+      this.exitFullscreen()
+    } else {
+      this.requestFullscreen()
+    }
+
+    this.setState({ fullscreen: !this.state.fullscreen })
+  }
+
+  exitFullscreen () {
+    if (document.exitFullscreen) {
+      document.exitFullscreen()
+    } else if (document.mozCancelFullScreen) {
+      document.mozCancelFullScreen()
+    } else if (document.webkitExitFullscreen) {
+      document.webkitExitFullscreen()
+    }
+  }
+
+  requestFullscreen () {
+    if (this.refs.streamPlayer.requestFullscreen) {
+      this.refs.streamPlayer.requestFullscreen()
+    } else if (this.refs.streamPlayer.mozRequestFullScreen) {
+      this.refs.streamPlayer.mozRequestFullScreen()
+    } else if (this.refs.streamPlayer.webkitRequestFullscreen) {
+      this.refs.streamPlayer.webkitRequestFullscreen()
+    } else if (this.refs.streamPlayer.msRequestFullscreen) {
+      this.refs.streamPlayer.msRequestFullscreen()
+    }
```

Comment:
Good call! I had to make some minor tweaks to get it to work, but that's so much cleaner. Thanks!